applications/StructuralMechanicsApplication/custom_conditions/displacement_shifted_boundary_condition.h

```diff
@@ -0,0 +1,175 @@
+    KRATOS_CLASS_INTRUSIVE_POINTER_DEFINITION(DisplacementShiftedBoundaryCondition);
+
+    ///@}
+    ///@name Life Cycle
+    ///@{
+
+    /// Constructor with geometry
+    DisplacementShiftedBoundaryCondition(
+        IndexType NewId,
+        Geometry<Node>::Pointer pGeometry);
+
+    /// Constructor with geometry and properties
+    DisplacementShiftedBoundaryCondition(
+        IndexType NewId,
+        Geometry<Node>::Pointer pGeometry,
+        Properties::Pointer pProperties);
+
+    /// Constructor with array of nodes and properties
+    DisplacementShiftedBoundaryCondition(
+        IndexType NewId,
+        const NodesArrayType& ThisNodes);
+
+    /// Copy constructor
+    DisplacementShiftedBoundaryCondition(DisplacementShiftedBoundaryCondition const& rOther) = delete;
+
+    /// Destructor
+    ~DisplacementShiftedBoundaryCondition() override;
+
+    ///@}
+    ///@name Operators
+    ///@{
+
+    /// Assignment operator
+    DisplacementShiftedBoundaryCondition& operator=(DisplacementShiftedBoundaryCondition const& rOther) = delete;
+
+    ///@}
+    ///@name Operations
+    ///@{
+
+    Condition::Pointer Create(
+        IndexType NewId,
+        NodesArrayType const& ThisNodes,
+        Properties::Pointer pProperties) const override;
+
+    Condition::Pointer Create(
+        IndexType NewId,
+        GeometryType::Pointer pGeom,
+        Properties::Pointer pProperties) const override;
+
+    void Initialize(const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLocalSystem(
+        MatrixType& rLeftHandSideMatrix,
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateLeftHandSide(
+        MatrixType& rLeftHandSideMatrix,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void CalculateRightHandSide(
+        VectorType& rRightHandSideVector,
+        const ProcessInfo& rCurrentProcessInfo) override;
+
+    void EquationIdVector(
+        EquationIdVectorType& rResult,
+        const ProcessInfo& rCurrentProcessInfo) const override;
+
+    void GetDofList(
+        DofsVectorType& ConditionalDofList,
+        const ProcessInfo& CurrentProcessInfo) const override;
+
+    int Check(const ProcessInfo& rCurrentProcessInfo) const override;
+
+    ///@}
+    ///@name Input and output
+    ///@{
+
+    /// Turn back information as a string.
+    std::string Info() const override;
+
+    /// Print information about this object.
+    void PrintInfo(std::ostream& rOStream) const override;
+
+    /// Print object's data.
+    void PrintData(std::ostream& rOStream) const override;
+
+    ///@}
+protected:
+    ///@name Protected Life Cycle
+    ///@{
+
+    // Internal default constructor for serialization
+    DisplacementShiftedBoundaryCondition();
+
+    ///@}
+    ///@name Protected Operations
+    ///@{
+
+
+    ///@}
+private:
+    ///@name Private Operations
+    ///@{
+
+    void CalculateBtransCProjectionLinearisation(
```

Comment:
On the other hand in the header the & are in the correct side

Reply:
Doxygen is missing

Reply:
Done.